### PR TITLE
migrate telemetry module from ploomber/ploomber

### PR DESCRIPTION
**Pull request by Wxl19980214**
Migrate telemetry, config, and some tests here.

Small changes:
Added a if check [here](https://github.com/ploomber/core/compare/main...Wxl19980214:migrate-tel?expand=1#diff-c0500c77703069eaa61a0c9b61f11d8bc9852794558c9efcd58e0b22d993c51cR38), otherwise it kept failing locally because theres no env variable. Maybe in CI is different? But I don't think adding a check hurts.

Added a mock version [here](https://github.com/ploomber/core/compare/main...Wxl19980214:migrate-tel?expand=1#diff-c0500c77703069eaa61a0c9b61f11d8bc9852794558c9efcd58e0b22d993c51cR388). I kept failing this test because my ploomber version is the latest? Maybe we need to mock it so it is not the latest.

Changed way of importing __version__ because we are using in multiple projects. 

Also enforce a package_name parameter in log_api so that we know which package calls the telemetry.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -18,6 +18,9 @@
 DEV = [
     'pytest',
     'flake8',
+    'click',
+    'pyyaml',
+    'posthog',
```
Comment (edublancas) [inline setup.py:23]: these 3 should be in the install_requires section since they're mandatory

File: src/ploomber_core/telemetry/telemetry.py [hunk 2]
```diff
@@ -0,0 +1,521 @@
...
+                                                 "elapsed_time")
+    return event_id, uid, action, client_time, elapsed_time
+
+
+def log_api(action, package_name, version, client_time=None,
+            total_runtime=None, metadata=None):
+    """
+    This function logs through an API call, assigns parameters if missing like
+    timestamp, event id and stats information.
+    """
```
Comment (edublancas) [inline src/ploomber_core/telemetry/telemetry.py:395]: please add docstring documenting to document new arguments (package_name and version)

File: src/ploomber_core/telemetry/telemetry.py [hunk 3]
```diff
@@ -0,0 +1,521 @@
...
+            posthog.capture(distinct_id=uid,
+                            event='install_success_indirect',
+                            properties=props)
+
+        posthog.capture(distinct_id=uid, event=action, properties=props)
+
+
+# NOTE: should we log differently depending on the error type?
+# NOTE: how should we handle chained exceptions?
+def log_call(action, pkn, ver, payload=False):
```
Comment (edublancas) [inline src/ploomber_core/telemetry/telemetry.py:475]: document pkg and ver

**Review by edublancas — CHANGES_REQUESTED** [review]

**Comment by edublancas** [comment]
I missed this two comments:

> Added a if check [here](https://github.com/ploomber/core/compare/main...Wxl19980214:migrate-tel?expand=1#diff-c0500c77703069eaa61a0c9b61f11d8bc9852794558c9efcd58e0b22d993c51cR38), otherwise it kept failing locally because theres no env variable. Maybe in CI is different? But I don't think adding a check hurts.

> Added a mock version [here](https://github.com/ploomber/core/compare/main...Wxl19980214:migrate-tel?expand=1#diff-c0500c77703069eaa61a0c9b61f11d8bc9852794558c9efcd58e0b22d993c51cR388). I kept failing this test because my ploomber version is the latest? Maybe we need to mock it so it is not the latest.

github is not allowing me to see this anymore, can you point me to the right files?

**Comment by idomic** [comment]
@Wxl19980214 ^^

**Pull request merged**